src/lib.rs

```diff
@@ -1,37 +1,304 @@
+//!
+//! # Examples
+//!
+//! * A basic implementation could look like this:
+//!     ```rust,no_run
+//!     use std::process;
+//!     use xtask_wasm::{anyhow::Result, clap};
+//!
+//!     #[derive(clap::Parser)]
+//!     struct Opt {
+//!         #[clap(subcommand)]
+//!         cmd: Command,
+//!     }
+//!
+//!     #[derive(clap::Parser)]
+//!     enum Command {
+//!         Dist(xtask_wasm::Dist),
+//!         Watch(xtask_wasm::Watch),
+//!         Serve(xtask_wasm::DevServer),
+//!     }
+//!
+//!
+//!     fn main() -> Result<()> {
+//!         let opt: Opt = clap::Parser::parse();
+//!
+//!         match opt.cmd {
+//!             Command::Dist(dist) => {
+//!                 let dist = dist
+//!                     .dist_dir_path("dist")
+//!                     .static_dir_path("project/static")
+//!                     .app_name("project")
+//!                     .run_in_workspace(true)
+//!                     .run("project")?;
+//!
+//!                 log::info!("Built at {}", dist.dist_dir.display());
+//!
+//!             }
+//!             Command::Watch(watch) => {
+//!                 let mut command = process::Command::new("cargo");
+//!                 command.args(["xtask", "dist"]);
+//!
+//!                 log::info!("Starting to watch");
+//!                 watch.exclude_workspace_path("dist").run(command)?;
+//!             }
+//!             Command::Serve(mut dev_server) => {
+//!                 let mut command = process::Command::new("cargo");
+//!                 command.args(["xtask", "dist"]);
+//!
+//!                 log::info!("Starting the dev server");
+//!                 dev_server.command(command).start("dist")?;
+//!             }
+//!         }
+//!
+//!         Ok(())
+//!     }
+//!     ```
+//!
+//! * [`examples/demo`](https://github.com/rustminded/xtask-wasm/tree/main/examples/demo)
+//!     provides an implementation of xtask-wasm to build the `webapp` package, an
+//!     `hello world` app using [Yew](https://yew.rs/). This example demonstrate a
+//!     simple directory layout and a customized build process that use the
+//!     [`wasm-opt`] feature.
+//!
+//! The available subcommands are:
+//!
+//! * Build the `webapp` package.
+//!     ```console
+//!     cargo xtask build
+//!     ```
+//!     * Build the `webapp` package, download the wasm-opt binary and optimize
+//!         the Wasm generated by the build process.
+//!         ```console
+//!         cargo xtask build --optimize
+//!         ```
+//!
+//! * Build the `webapp` package and watch for changes in the workspace root.
+//!     ```console
+//!     cargo xtask watch
+//!     ```
+//!
+//! * Serve an optimized `webapp` dist on `127.0.0.1:8000` and watch for
+//!     changes in the workspace root.
+//!     ```console
+//!     cargo xtask serve
+//!     ```
+//!
+//! Additional flags can be found using `cargo xtask <subcommand> --help`
+//!
+//! # Features
+//!
+//! * wasm-opt - Enable `WasmOpt` that download the
```

Comment:
You know there are typography rules for using a dash like that. The easiest way to make it correct is to not try to write the em/en dash (or read the manual: https://en.wikipedia.org/wiki/Dash)

Here you must use backticks around wasm-opt because it is the actual technical name of the feature that the user should use precisely.

```suggestion
//! * `wasm-opt`: enable the [`WasmOpt`] struct that helps downloading and using wasm-opt very easily.
```